third_party/terraform/resources/resource_google_billing_subaccount.go

```diff
@@ -0,0 +1,134 @@
+package google
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"google.golang.org/api/cloudbilling/v1"
+)
+
+func resourceBillingSubaccount() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceBillingSubaccountCreate,
+		Read:   resourceBillingSubaccountRead,
+		Delete: resourceBillingSubaccountDelete,
+		Update: resourceBillingSubaccountUpdate,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"display_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"open": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"master_billing_account": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					if new == canonicalBillingAccountName(old) {
+						return false
+					}
+					return true
+				},
+			},
+			"billing_account_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"rename_on_destroy": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+		},
+	}
+}
+
+func resourceBillingSubaccountCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	billingAccount := new(cloudbilling.BillingAccount)
```

Comment:
```suggestion
	billingAccount := &cloudbilling.BillingAccount{}
```

For cases where you used `new`, struct initialization is more consistent with the rest of the codebase.